lib/portage/eapi.py

```diff
@@ -324,113 +203,124 @@ def eapi_has_sysroot(eapi):
 )
 
 
-_eapi_attr_func_prefixes = (
-    "eapi_allows_",
-    "eapi_has_",
-    "eapi_requires_",
-    "eapi_supports_",
-    "eapi_",
-)
-
-
-def _eapi_func_decorator(func, attr_getter):
-    def wrapper(eapi):
-        return attr_getter(_get_eapi_attrs(eapi))
-
-    wrapper.func = func
-    wrapper.__doc__ = func.__doc__
-    return wrapper
-
-
-def _decorate_eapi_funcs():
-    """
-    Decorate eapi_* functions so that they use _get_eapi_attrs(eapi)
-    to cache results.
-    """
-    decorated = {}
-    for k, v in globals().items():
-        if not (
-            isinstance(v, types.FunctionType) and k.startswith(_eapi_attr_func_prefixes)
-        ):
-            continue
-        for prefix in _eapi_attr_func_prefixes:
-            if k.startswith(prefix):
-                attr_name = k[len(prefix) :]
-                if hasattr(_eapi_attrs, attr_name):
-                    decorated[k] = _eapi_func_decorator(
-                        v, operator.attrgetter(attr_name)
-                    )
-                    break
-    globals().update(decorated)
-
-
-_decorate_eapi_funcs()
-
-
-_eapi_attrs_cache = {}
-
-
-def _get_eapi_attrs(eapi):
+class Eapi:
+    ALL_EAPIS = (
+        "0",
+        "1",
+        "2",
+        "3",
+        "4",
+        "4-python",
+        "4-slot-abi",
+        "5",
+        "5-progress",
+        "6",
+        "7",
+        "8",
+    )
+    EAPI_RANGE_MAP = {
```

Comment:
This EAPI_RANGE_MAP variable seems to be unused.

Reply:
You're right, this is an artifact of an older implementation. I'll take it out.